fvm/src/kernel/default.rs

```diff
@@ -539,6 +541,19 @@ where
             .call_manager
             .charge_gas(self.call_manager.price_list().on_verify_post(verify_info))?;
 
+        #[allow(clippy::collapsible_if)]
+        if self.call_manager.context().network.chain_id == ChainID::from(314159) {
+            if self.call_manager.context().epoch <= 498691
+                && verify_info.proofs[0].post_proof == StackedDRGWindow32GiBV1P1
+            {
+                let mut verify_info_copy = verify_info.clone();
```

Comment:
Other efforts to fail early, such as returning Err here, or panicking without calling `verify_post` lead to slightly different gas usage. This surprises me, since these operations seem to be uncharged, but :shrug:

Reply:
I looked at the code and thoroughly convinced myself it did not, but...you're right. That's much nicer.